Maple2.Database/Storage/Metadata/QuestMetadataStorage.cs

```diff
@@ -35,6 +36,23 @@ public class QuestMetadataStorage : MetadataStorage<int, QuestMetadata>, ISearch
         return true;
     }
 
+    public IEnumerable<QuestMetadata> GetQuests() {
+        lock (Context) {
+            return Context.QuestMetadata
+                .ToList();
+        }
+    }
+
+    public IEnumerable<QuestMetadata> GetQuestsByNpc(int npcId) {
+        lock (Context) {
+            // this works but seems questionable?? Cannot search before AsEnumerable
```

Comment:
Yes this is bad. There is a query to do this `JSON_EXTRACT(Basic, '$.StartNpc')=...`.

I assume you already tried:
```cs
Context.QuestMetadata
    .Where(quest => quest.Basic.StartNpc == npcId)
    .ToList();
```

You can try a raw SQL query instead:
```cs
Context.QuestMetadata
    .FromSqlRaw($"SELECT * FROM `quest` WHERE JSON_EXTRACT(Basic, '$.StartNpc')={npcId}")
    .ToList();
```